### PR TITLE
Create binary_sensor.xiaomi_gw.markdown

### PR DESCRIPTION
**Pull request by jcastro**
Initial commit for binary_sensors sub-component of xiaomi_gw

**Description:**

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#<home-assistant PR number goes here>

### CODE REVIEW

File: source/_components/binary_sensor.xiaomi_gw.markdown [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        from: 'off'
+        to: 'on'
+      action:
+        - service: notify.html5
+          data:
+            title: Fire alarm!
+            message: Fire/Smoke detected!
+        - service: xiaomi.play_ringtone
+          data:
+            gw_sid: xxxxxxxxxxxx
```
Comment (Danielhiversen) [inline source/_components/binary_sensor.xiaomi_gw.markdown]: Should now be gw_mac

File: source/_components/binary_sensor.xiaomi_gw.markdown [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+      trigger:
+        platform: event
+        event_type: click
+        event_data:
+          entity_id: binary_sensor.switch_158d000xxxxxc2
+          click_type: long_click_press
+      action:
+        service: xiaomi.play_ringtone
+        data:
+          gw_sid: xxxxxxxxxxxx
```
Comment (Danielhiversen) [inline source/_components/binary_sensor.xiaomi_gw.markdown]: Should now be gw_mac

Comment (jcastro) [inline source/_components/binary_sensor.xiaomi_gw.markdown]: thanks! fixed 👍

**Comment by syssi** [comment]
Do you like to add the following example automation for the cube?

- Cube
```yaml
    # Trigger for a few cube events

    - alias: Cube event flip90
      trigger:
        platform: event
        event_type: cube_action
        event_data:
          entity_id: binary_sensor.cube_15xxxxxxxxxxxx
          action_type: flip90
      action:
        - service: light.turn_on
          entity_id: light.gateway_light_28xxxxxxxxxx
          data:
            color_name: "springgreen"
                                                                                                                                                                                                       
    - alias: Cube event flip180                                                                                                                                                                              
      trigger:                                                                                                                                                                                         
        platform: event                                                                                                                                                                                
        event_type: cube_action                                                                                                                                                                        
        event_data:                                                                                                                                                                                    
          entity_id: binary_sensor.cube_15xxxxxxxxxxxx                                                                                                                                                 
          action_type: flip180                                                                                                                                                                         
      action:                                                                                                                                                                                          
        - service: light.turn_on
          entity_id: light.gateway_light_28xxxxxxxxxx
          data:
            color_name: "darkviolet"
                                                                                                                                                                                                       
    - alias: Cube event move                                                                                                                                                                                 
      trigger:                                                                                                                                                                                         
        platform: event                                                                                                                                                                                
        event_type: cube_action                                                                                                                                                                        
        event_data:                                                                                                                                                                                    
          entity_id: binary_sensor.cube_15xxxxxxxxxxxx                                                                                                                                                 
          action_type: move                                                                                                                                                                            
      action:                                                                                                                                                                                          
        - service: light.turn_on
          entity_id: light.gateway_light_28xxxxxxxxxx
          data:
            color_name: "gold"
                                                                                                                                                                                                       
    - alias: Cube event tap_twice                                                                                                                                                                            
      trigger:                                                                                                                                                                                         
        platform: event                                                                                                                                                                                
        event_type: cube_action                                                                                                                                                                        
        event_data:
          entity_id: binary_sensor.cube_15xxxxxxxxxxxx
          action_type: tap_twice
      action:
        - service: light.turn_on
          entity_id: light.gateway_light_28xxxxxxxxxx
          data:
            color_name: "deepskyblue"
    
    - alias: Cube event shake_air
      trigger:
        platform: event
        event_type: cube_action
        event_data:
          entity_id: binary_sensor.cube_15xxxxxxxxxxxx
          action_type: shake_air
      action:
        - service: light.turn_on
          entity_id: light.gateway_light_28xxxxxxxxxx
          data:
            color_name: "blue"
```

File: source/_components/binary_sensor.xiaomi_gw.markdown [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+          data:
+            title: Fire alarm!
+            message: Fire/Smoke detected!
+        - service: xiaomi.play_ringtone
+          data:
+            gw_mac: xxxxxxxxxxxx
+            ringtone_id: 2
+            ringtone_vol: 100
+  ```
+- Smoke
```
Comment (syssi) [inline source/_components/binary_sensor.xiaomi_gw.markdown]: Please move the headline "Smoke" above the "Send notification on fire alarm" automation. :-)

File: source/_components/binary_sensor.xiaomi_gw.markdown [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+- Gas
```
Comment (syssi) [inline source/_components/binary_sensor.xiaomi_gw.markdown]: - Gas
```yaml
    # Trigger for natgas detected

    - alias: Send notification on gas alarm
      trigger:
        platform: state
        entity_id: binary_sensor.natgas_sensor_158dxxxxxxxxxx
        from: 'off'
        to: 'on'
      action:
        - service: notify.html5
          data_template:
            title: Gas alarm!
            message: 'Gas with a density of {{ states.binary_sensor.natgas_sensor_158dxxxxxxxxxx.attributes.density }} detected.'
```

File: source/_components/binary_sensor.xiaomi_gw.markdown [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+- Xiaomi Wireless Button (available events are `single`, `double`, `hold`, `long_click_press` and `long_click_release`).
+ ```yaml
```
Comment (syssi) [inline source/_components/binary_sensor.xiaomi_gw.markdown]: The new/square version of the Xiaomi Wireless Button just supports `single` and `double` click. Furthermore the space between two clicks to generate a double click must be quite large now.

Comment (jcastro) [inline source/_components/binary_sensor.xiaomi_gw.markdown]: thanks! everything fixed!

**Comment by syssi** [comment]
The wireless wall switch (86sw1, 86sw2) is the last missing binary_sensor. I will provide a few lines tomorrow.

**Comment by syssi** [comment]
- Aqara Wireless Switch

The Aqara Wireless Switch is available as single-key and double-key version. Each key behaves like the Wireless Button limited to the click event `single`. The double key version adds a third device called binary_sensor.wall_switch_both_158xxxxxxxxx12 which reports a click event called `both` if both keys are pressed.


```yaml
    - alias: Decrease brightness of the gateway light                                                                                                                                                                                                                              
      trigger:                                                                                                                                                                                                                                                                 
        platform: event                                                                                                                                                                                                                                                        
        event_type: click                                                                                                                                                                                                                                                      
        event_data:                                                                                                                                                                                                                                                            
          entity_id: binary_sensor.wall_switch_left_158xxxxxxxxx12                                                                                                                                                                                                             
          click_type: single                                                                                                                                                                                                                                                   
      action:                                                                                                                                                                                                                                                                  
        service: light.turn_on                                                                                                                                                                                                                                                 
        entity_id: light.gateway_light_34xxxxxxxx13                                                                                                                                                                                                                            
        data_template:                                                                                                                                                                                                                                                         
          brightness: >-
            {% if states.light.gateway_light_34xxxxxxxx13.attributes.brightness %}
              {% if states.light.gateway_light_34xxxxxxxx13.attributes.brightness - 60 >= 10 %}
                {{states.light.gateway_light_34xxxxxxxx13.attributes.brightness - 60}}
              {% else %}
                {{states.light.gateway_light_34xxxxxxxx13.attributes.brightness}}
              {% endif %}
            {% else %}
              10
            {% endif %}
    
    - alias: Increase brightness of the gateway light
      trigger:
        platform: event
        event_type: click
        event_data:
          entity_id: binary_sensor.wall_switch_right_158xxxxxxxxx12                                                                                                                                                                                                             
          click_type: single
      action:
        service: light.turn_on
        entity_id: light.gateway_light_34xxxxxxxx13
        data_template:
          brightness: >-
            {% if states.light.gateway_light_34xxxxxxxx13.attributes.brightness %}
              {% if states.light.gateway_light_34xxxxxxxx13.attributes.brightness + 60 <= 255 %}
                {{states.light.gateway_light_34xxxxxxxx13.attributes.brightness + 60}}
              {% else %}
                {{states.light.gateway_light_34xxxxxxxx13.attributes.brightness}}
              {% endif %}
            {% else %}
              10
            {% endif %}
    
    - alias: Turn off the gateway light
      trigger:
        platform: event
        event_type: click
        event_data:
          entity_id: binary_sensor.wall_switch_both_158xxxxxxxxx12                                                                                                                                                                                                             
          click_type: both
      action:
        service: light.turn_off
        entity_id: light.gateway_light_34xxxxxxxx13
```

**Comment by jcastro** [comment]
Oh nice, do you think it will be possible to also support the award wall switch with its buttons set to wireless mode? (as if shows on the Mi home app preferences for the switch)

**Comment by syssi** [comment]
@jcastro Unfortunately, no. The device doesn't fire events in decoupled mode at the moment. I will implement the feature as soon a new gateway firmware introduces a new behaviour.

**Pull request merged**